lib/knock/users.rb

```diff
@@ -317,6 +317,22 @@ def set_channel_data(id:, channel_id:, channel_data:)
         execute_request(request: request)
       end
 
+      # Unsets user's channel data for the given channel id
+      #
+      # @param [String] id the user ID
+      # @param [String] channel_id target channel ID
+      # @param [Hash] channel_data channel data
+      #
+      # @return [Hash] channel data
+      def unset_channel_data(id:, channel_id:)
```

Comment:
This one is failing for me:

```
irb(main):025:0> Knock::Objects.unset_channel_data(collection: "asset", id: "1", channel_id: "21eb901c-54f6-44ca-a7e9-4f4503d60d36")
/Users/juanazam/.asdf/installs/ruby/3.0.2/lib/ruby/3.0.0/json/common.rb:216:in `initialize': no implicit conversion of nil into String (TypeError)
```

Reply:
I think it fails because the delete request doesn't contain any body on the response

Reply:
🤔 that means our other delete requests are failing too. Will fix.

Reply:
updating https://github.com/knocklabs/knock-ruby/blob/main/lib/knock/client.rb#L21 to

```
JSON.parse(response.body) if response.body
```

Does the trick

Reply:
@cjbell yep, they are